extensions/git/src/repository.ts

```diff
@@ -1581,7 +1581,16 @@ export class Repository implements Disposable {
 	}
 
 	async checkoutTracking(treeish: string, opts: { detached?: boolean } = {}): Promise<void> {
```

Comment:
It feels like this should be an either/or scenario.

If the user wants to checkout detached, that means they don't want to create a new branch (tracking or otherwise). If the user wants to checkout a tracking branch, then it's not detached, by definition.

If I read it correctly, whenever the user wants to checkout detached, this PR will (1) create a tracking branch and (2) detach HEAD from it. Having that branch created would be unexpected, right?

Reply:
Your assessment is correct. I will explore implementing a different fix that does not end up creating a tracking branch.